static/app/views/organizationIntegrations/integrationExternalTeamMappings.tsx

```diff
@@ -19,15 +18,19 @@ type Props = AsyncComponent['props'] &
 
 type State = AsyncComponent['state'] & {
   teams: Team[];
-  queryResults: Team[];
+  queryResults: {
```

Comment:
Okay, this is pretty confusing. So the endpoint we submit to for teams is dependant on the team we're submitting with. The problem is, the form data we submit only has the `teamId`, but the endpoint is based on the `teamSlug`. That means we need to save the results that we took `teamId` from, and parse the `teamSlug` out of it. That's what `onResults` is useful for.

When that fires, we write to the this higher-order component's state with the results of the user's query. Then, when they submit, we use that query response we stored in state, extract the slug, and provide the endpoint in this class's `getBaseFormEndpoint` method.

Prior, we only every had one state key since only one edit could be performed at a time. If the user was editing a team mapping, the modal would open, they'd type some text, and we'd store those results here. Now, we store both edits from the modal, and edits from the inline forms, meaning we need a unique state key  for all of them, which I refer to as the `mappingKey`.

So, in practice we can go through an example:

User has 3 teams `@team1`, `@team2` and `@team3`. If they make a change to `@team1`'s mapping, we store the results from them typing in the inline field in our state on `this.state.queryResults.team1`. When they blur the field and it submits, we read from that state to get the new team slug, and update the mapping.

Let's now say they want to add a new `@team4`. They would open the modal and type out a "@team4" as the externalName. Since we don't want to use this user generated text as a team mapping (since right after it'd also have an inline form) we treat the modal differently and use a static mappingKey for the modal. The results after the user types to map `@team4` are stored at `this.state.queryResults.MODAL_RESULTS` (See line 89)